backend/src/utils/route-security.ts

```diff
@@ -257,24 +257,36 @@ export const sanitizeNotebookForSecurity = async <
 
   // Container based items
   secureNotebook?.spec?.template?.spec?.containers?.forEach((container) => {
-    // Secrets & ConfigMaps
+    // Configmaps
     container.env?.forEach((env) => {
-      if (env.name.startsWith('jupyterhub-singleuser-profile-')) {
-        // Env var for a configmap or secret we generated
-        const allowedValue = `jupyterhub-singleuser-profile-${translatedUsername}-env`;
-        if (env.name !== allowedValue) {
+      if (env.valueFrom?.configMapKeyRef?.name?.startsWith('jupyterhub-singleuser-profile-')) {
+        // Env var for a configmap
+        const allowedValue = `jupyterhub-singleuser-profile-${translatedUsername}-envs`;
+        if (env.valueFrom?.configMapKeyRef?.name !== allowedValue) {
           // Was not targeted at their user
           fastify.log.warn(
-            `${username} submitted a Notebook that contained an env (${env.name}) that was not for them. Reset back to them.`,
+            `${username} submitted a Notebook that contained an env configmap (${env.valueFrom?.configMapKeyRef?.name}) that was not for them. Reset back to them.`,
```

Comment:
```suggestion
            `${username} submitted a Notebook that contained an env configmap (${env.valueFrom.configMapKeyRef.name}) that was not for them. Reset back to them.`,
```
FWIW, you have already verified it exists with your if statement, you don't need optionals anymore.

Reply:
yeah, my bad, i didn't re-checked the code since I need to go, changing it right now